helix-term/src/ui/prompt.rs

```diff
@@ -80,20 +85,29 @@ impl Prompt {
         self.exit_selection();
     }
 
-    pub fn change_completion_selection(&mut self) {
+    pub fn change_completion_selection(&mut self, direction: CompletionDirection) {
         if self.completion.is_empty() {
             return;
         }
-        let index = self.selection.map_or(0, |i| i + 1) % self.completion.len();
+
+        let index = match direction {
+            CompletionDirection::Forward => {
+                self.selection.map_or(0, |i| i + 1) % self.completion.len()
+            }
+            CompletionDirection::Backward => {
+                (self.selection.unwrap_or(0) + self.completion.len() - 1) % self.completion.len()
+            }
+        };
```

Comment:
```suggestion
        let index = match direction {
            CompletionDirection::Forward => {
                self.selection.map_or(0, |i| i + 1)
            }
            CompletionDirection::Backward => {
                self.selection.unwrap_or(0) + self.completion.len() - 1
            }
        } % self.completion.len();
```
I think we can do this instead.

Reply:
Thanks for that! 🎉